src/Microsoft.AspNetCore.Hosting/ConfigureContainerBuilder.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Hosting.Internal
+{
+    public class ConfigureContainerBuilder
+    {
+        public ConfigureContainerBuilder(MethodInfo configureContainerMethod)
+        {
+            MethodInfo = configureContainerMethod;
+        }
+
+        public MethodInfo MethodInfo { get; }
+
+        public Action<object> Build(object instance) => container => Invoke(instance, container);
+
+        private void Invoke(object instance, object container)
+        {
+            if (MethodInfo == null)
+            {
+                return;
+            }
+
+            // Only support IServiceCollection parameters
+            var parameters = MethodInfo.GetParameters();
+            if (parameters.Length != 1 ||
+                parameters.Any(p => !p.ParameterType.IsAssignableFrom(container.GetType())))
+            {
+                throw new InvalidOperationException($"The {MethodInfo.Name} method must either be parameterless or take only one parameter of type {container.GetType()}.");
```

Comment:
Isn't this message wrong? The check is `parameters.Length != 1`, which means it cannot be parameterless. Having a parameterless version of this method makes no sense, IMO.